src/main/java/com/example/bulletinboard/config/WebSecurityConfig.java

```diff
@@ -20,7 +20,7 @@ public class WebSecurityConfig {
             "/webjars/**",
             "/login",
             "/register",
-            "/ads/**",
+            "/ads/",
```

Comment:
```suggestion
            "/ads",
```
у тебя нет пути /ads/, есть /ads